### PR TITLE
Oleksii_Gerasymov Homework Module 3: Exceptions and IO streams

### PR DESCRIPTION
**Pull request by Doozer-Dnepr**

### CODE REVIEW

File: Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/.idea/compiler.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      <entry name="!?*.clj" />
+      <entry name="!?*.aj" />
+    </wildcardResourcePatterns>
+    <annotationProcessing>
+      <profile default="true" name="Default" enabled="false">
+        <processorPath useClasspath="true" />
+      </profile>
+    </annotationProcessing>
+  </component>
+</project>
```
Comment (technicaltaskjava) [inline Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/.idea/compiler.xml]: use .gitignore file to do not add configuration and settings files.

File: Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/.idea/misc.xml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    <OptionsSetting value="true" id="Update" />
+    <OptionsSetting value="true" id="Status" />
+    <OptionsSetting value="true" id="Edit" />
+    <ConfirmationsSetting value="0" id="Add" />
+    <ConfirmationsSetting value="0" id="Remove" />
+  </component>
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+    <output url="file://$PROJECT_DIR$/out" />
+  </component>
+</project>
```
Comment (technicaltaskjava) [inline Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/.idea/misc.xml]: use .gitignore file to do not add configuration and settings files.

File: Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/.idea/workspace.xml [hunk 3]
```diff
@@ -0,0 +1,1313 @@
...
+    </entry>
+    <entry file="file://$PROJECT_DIR$/src/main/java/com/epam/t05/FilmsCollection.java">
+      <provider selected="true" editor-type-id="text-editor">
+        <state vertical-scroll-proportion="0.0">
+          <caret line="40" column="8" selection-start-line="40" selection-start-column="8" selection-end-line="40" selection-end-column="14" />
+          <folding>
+            <element signature="e#281#282#0" expanded="true" />
+            <element signature="e#309#310#0" expanded="true" />
+            <element signature="e#351#352#0" expanded="true" />
+
```
Comment (technicaltaskjava) [inline Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/.idea/workspace.xml]: Use .gitignore.

File: Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/StreamsAndExceptions.java [hunk 4]
```diff
@@ -0,0 +1,66 @@
+package main.java.com.epam;
```
Comment (technicaltaskjava) [inline Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/StreamsAndExceptions.java]: do not use main.java as it's default build path.

File: Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/t01/WorkingWithTxtFiles.java [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                    }
+                }
+            }
+            catch (NumberFormatException userException) {
+                System.out.println("Incorrect input. You must input digit from 1 to 6.\n");
+            }
+        }
+    }
+
+}
```
Comment (technicaltaskjava) [inline Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/t01/WorkingWithTxtFiles.java]: Task 1 looks good.

File: Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/t02/WorkWithPropertyFiles.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    public static void mainMenu() {
+
+        Properties propertyFile = null;
+
+        boolean returnToMainMenu = false;
+        while (returnToMainMenu == false) {
+            System.out.println("Current folder is: "+WorkWithFolder.currentFolder);
+            System.out.println("What do you want to do:");
+            System.out.println("[1] Print out current directory.");
+            System.out.println("[2] Change directory.");
```
Comment (technicaltaskjava) [inline Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/t02/WorkWithPropertyFiles.java]: Impossible to change directory second time

File: Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/t02/WorkWithPropertyFiles.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    public static String searchPropertyInFile(Properties propertyFile, String currentLineToSearch) throws IOException {
+                    String resultValue = propertyFile.getProperty(currentLineToSearch);
+                    if (resultValue != null) {
+                        return resultValue;
+                    }
+                    else {
+                        throw new IOException("Property not found");
+                    }
+    }
+}
```
Comment (technicaltaskjava) [inline Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/t02/WorkWithPropertyFiles.java]: Second task is ok.

File: Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/t04/WorkWithCharIO.java [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+                currentWritter.write(Integer.toString(entranceArray[indexArray]));
+                currentWritter.write(System.getProperty("line.separator"));
+            }
+            currentWritter.close();
+        }
+        else {
+            throw new IOException("File already exists");
+        }
+    }
+}
```
Comment (technicaltaskjava) [inline Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/t04/WorkWithCharIO.java]: 4-th task looks ok.

File: Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/t05/FilmFactory.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    public static void mainMenu() {
+
+        FilmsCollection myFilms = new FilmsCollection();
+
+        boolean returnToMainMenu = false;
+
+        while (returnToMainMenu == false) {
+                System.out.println("Choose the action:");
+                System.out.println("[1] Show full collection.");
+                System.out.println("[2] Restore collection.");
```
Comment (technicaltaskjava) [inline Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/main/java/com/epam/t05/FilmFactory.java]: Impossible to restore collection. Exception is appeared.

2
Exception in thread "main" java.lang.ClassCastException: main.java.com.epam.t05.FilmsCollection cannot be cast to com.epam.t05.FilmsCollection
	at com.epam.t05.FilmsCollection.restoreFilms(FilmsCollection.java:33)
	at com.epam.t05.FilmFactory.mainMenu(FilmFactory.java:47)
	at com.epam.StreamsAndExceptions.main(StreamsAndExceptions.java:48)

File: Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/test/java/com/epam/TestFileCreateAndDelete.java [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+    @Test
+    public void createAndDeleteTxtFileTest() throws Exception {
+        String testPath = WorkWithFile.getCurrentFilePath("temp");
+        WorkWithFile.createTxtFile("temp");
+        File currentFile = new File(testPath + ".txt");
+        assertEquals(true, currentFile.exists());
+        WorkWithFile.deleteTxtFile("temp");
+        assertEquals(false, currentFile.exists());
+    }
+}
```
Comment (technicaltaskjava) [inline Oleksii_Gerasymov/Module 3 - Exceptions and IO streams/src/test/java/com/epam/TestFileCreateAndDelete.java]: Only one task is covered by tests.

**Comment by Doozer-Dnepr** [comment]
1. About .gitignore - checked. Mistake removed at fourth task.
2. About using main.* at buliding main pass - checked. Mistake removed at
fourth task.
3. About changing directory - I'll try to debug. Everything was good but...
Checked.
4. About restoring collection of films - I'll try to debug. While testing i
saw that everything was good. I'll debug this problem.
5. About test coverage - There were no any task to cover all method by
tests. Teacher asked about writnig several tests as example and run it. But
OK. Checked.

Thank you for supporting.

2016-03-19 19:17 GMT+02:00 technicaltaskjava <notifications@github.com>:

**Pull request merged**